### ISSUE TITLE
Where does "Block" live?

### ISSUE THREAD

**Issue by mikeal** [ipld/ipld#47]
Out of necessity, we have implementations of "Block" level objects and interfaces in IPFS. This includes an HTTP API for block level access and block level objects like `require('ipfs-block')`. 

However, Blocks are a necessary IPLD level concept. Any documentation or tutorial about IPLD ends up needing to define a block as a CID + data. 

In a practical sense, you can't build much on IPLD w/o a Block object unless you're going to use [cid, data] tuples. As we imagine an ecosystem on top of IPLD that works w/o IPFS we *need* a Block object. In fact, some of the libraries I've already written using only IPLD end up requiring `ipfs-block`.

I'd like to potentially move these implementations and docs to the IPLD layer/org and write up more uniform definitions. I hope that this can also clear up some of the confusion that leads to [issues like this](https://github.com/ipld/js-ipld/issues/149) and put us on a path towards future block services that operate without a dependence on IPFS (or IPFS repo).

**Comment by vmx** [comment]
I agree with @mikeal. I just like to mention that I removed the "Block" from the ipld-format spec (https://github.com/ipld/interface-ipld-format/issues/21). I think a "Block" should only be used if both CID and data are needed, not if only one of them is (especially for low level interfaces like IPLD Format)

**Comment by alanshaw** [comment]
I'm 👍 on moving it. For reference:

> Package blocks contains the lowest level of IPLD data structures.
> https://github.com/ipfs/go-block-format/blob/master/blocks.go#L1

So it seems to also be considered an IPLD thing by go-ipfs already (cc @Stebalien).

* [ ] There should be a corresponding initiative to https://github.com/ipld/js-ipld-block/issues/43 for moving `go-block-format` to IPLD
